server/src/main/java/org/opensearch/common/blobstore/AsyncMultiStreamEncryptedBlobContainer.java

```diff
@@ -44,12 +46,8 @@ public void asyncBlobUpload(WriteContext writeContext, ActionListener<Void> comp
 
     @Override
     public void readBlobAsync(String blobName, ActionListener<ReadContext> listener) {
-        throw new UnsupportedOperationException();
-    }
-
-    @Override
-    public void asyncBlobDownload(String blobName, Path fileLocation, ThreadPool threadPool, ActionListener<String> completionListener) {
```

Comment:
ThreadPool needs to be removed from this abstraction.

Reply:
Also, why do we have two versions of async read - readBlobAsync and asyncBlobDownload?

Reply:
Discussed this offline.